harness/determined/exec/tensorboard.py

```diff
@@ -1,143 +1,250 @@
-    """
-    Gets the version of the tensorboard package currently installed. Used
-    by downstream processes to determine args passed in.
-    :return: version in the form of (major, minor) tuple
-    """
+    except (ConnectionError, HTTPError) as exp:
+        logger.warning(f"OK: Tensorboard not responding to HTTP: {exp}")
 
-    major, minor, _ = version.split(".")
+    except ValueError as exp:
+        logger.warning(str(exp))
 
-    return major, minor
+    except json.JSONDecodeError as exp:
+        logger.warning(f"OK: Could not JSONDecode Tensorboard HTTP response: {exp}")
 
+    if len(tags) != 0 and any([len(v) for v in tags.values()]):
+        print(TENSORBOARD_TRIGGER_READY_MSG)
+        return True
 
-def get_tensorboard_args(args: List[str]) -> List[str]:
-    """
-    Builds tensorboard startup args from args passed in from tensorboard-entrypoint.sh
-    Args are added and deprecated at the mercy of tensorboard; all of the below are necessary to
-    support versions 1.14, 2.4, and 2.5
+    return False
 
-    - If multiple directories are specified and the tensorboard version is > 1,
-    use legacy logdir_spec behavior
 
-    - Tensorboard 2+ no longer exposes all ports. Must pass in "--bind_all" to expose localhost
+def start_tensorboard(tb_version, paths, add_tb_args):
 
-    - Tensorboard 2.5.0 introduces an experimental feature (default load_fast=true)
-    which prevents multiple plugins from loading correctly.
-    """
-    task_id = os.environ["DET_TASK_ID"]
-    port = os.environ["TENSORBOARD_PORT"]
+    # Create local temporary directory
+    script_dir = os.path.dirname(__file__)
+    temp_dir = os.path.join(script_dir, f"tb_events_{str(uuid.uuid4())}")
+    os.makedirs(temp_dir, mode=0o777, exist_ok=True)
 
-    version = args.pop(0)
+    # Get fetcher and perform initial fetch
+    fetcher = build_storage_fetcher(temp_dir)
+    fetcher.download_new_files()
 
-    # logdir is the second argument passed in from tensorboard_manager.go. If multiple directories
-    # are specified and the tensorboard version is > 1, use legacy logdir_spec behavior. NOTE:
-    # legacy logdir_spec behavior is not supported by many tensorboard plugins
-    logdir = args.pop(0)
+    # Build Tensorboard args and launch process.
+    tb_args = get_tensorboard_args(tb_version, temp_dir, add_tb_args)
+    logger.debug(f"tensorboard args: {tb_args}")
+    tensorboard_process = subprocess.Popen(tb_args)
 
-    tensorboard_args = ["tensorboard", f"--port={port}", f"--path_prefix=/proxy/{task_id}", *args]
+    # Loop state.
+    stop_time = time.time() + 600
+    tb_has_metrics = False
 
-    major, minor = get_tensorboard_version(version)
+    try:
+        while True:
+            # Check if tensorboard process is still alive.
+            ret_code = tensorboard_process.poll()
+            if ret_code is not None:
+                raise RuntimeError(f"Tensorboard process died, exit code({ret_code}).")
```

Comment:
instead of raising an error, just log this message and return `ret_code`.  Otherwise the outer `sys.exit(ret)` is pretty meaningless